### PR TITLE
Styling and UX improvements.

### PR DESCRIPTION
**Pull request by shannonwells**
## Styling and UX improvements
* Adds the Tailwind CSS framework as a basis for some lightweight styling using Heather's theme
* Changes where and how some things are displayed
* Also fixes the provider name display.

References #41 

### Details
* Made the connection display into its own Connection component like Provider and Capacity so they could be styled collectively.
* ProviderActions now hosts the action buttons, so they appear at the bottom. This made more sense from a behavioral point of view, but also, rendering more than one button at a time looked pretty ugly.  So the top section is just for status and no actions.  Associated unit tests were updated.
* The action form and the transaction status get a "card" treatment and are flex-displayed.
* Displaying Capacity with 'CAP' units for lack of a better term. Requesting input.  Also yes there is a bug with Capacity units vs. token units which I'd like to address separately. I just wanted it to display with SI prefixes.

I'd like to get this in to main before it goes any further since this was supposed to be "basic" styling and it's kind of a lot.

Screenshot 2023-08-14 at 3 28 27 PM

### CODE REVIEW

File: src/components/Capacity.svelte [hunk 1]
```diff
@@ -49,20 +54,31 @@
...
-  <h3>Capacity at Block {blockNumber}, Epoch {epochNumber}</h3>
-  <p>Provider name: {msaInfo.providerName}</p>
-  <p><strong>Remaining:</strong> {capacityDetails?.remainingCapacity}</p>
-  <p><strong>Total Issued:</strong> {capacityDetails?.totalCapacityIssued}</p>
-  <p><strong>Last replenished:</strong> Epoch {capacityDetails?.lastReplenishedEpoch}</p>
-  <p><strong>Staked Token:</strong> {capacityDetails?.totalCapacityIssued} {token}</p>
+<div class="pl-6 ml-6 border-l-8 border-aqua">
+  <h3 class="text-aqua font-bold">Capacity</h3>
+  {#if !connected}
+    <p>Not connected</p>
```
Comment (shannonwells) [inline src/components/Capacity.svelte:72]: All the "status" items now display more or less the same information until you are connected and we know if you are a Provider or you just have an MSA.

File: src/components/Stake.svelte [hunk 2]
```diff
@@ -81,7 +78,7 @@
         if (isFunction(thisWeb3FromSource) && isFunction(thisWeb3Enable)) {
           const extensions = await thisWeb3Enable('Frequency parachain provider dashboard: Adding Keys');
           if (extensions.length !== 0) {
-            const injectedExtension = await thisWeb3FromSource(signingKeys.meta.source);
+            const injectedExtension = await thisWeb3FromSource(signingKeys.meta.source.toString());
```
Comment (shannonwells) [inline src/components/Stake.svelte:81]: because typescript

File: src/routes/+page.svelte [hunk 3]
```diff
@@ -26,51 +28,30 @@
...
   const onChangeTxnSigningAddress = (evt: Event) => {
     let option = evt.target as HTMLOptionElement;
-    transactionSigningAddress.update((addr) => (addr = option.value));
+    storeMsaInfo.set({ isProvider: false, msaId: 0, providerName: '' });
+    transactionSigningAddress.set(option.value);
   };
 </script>
 
-<svelte:head>
-  <link rel="stylesheet" href="https://unpkg.com/mvp.css@1.12/mvp.css" />
```
Comment (shannonwells) [inline src/routes/+page.svelte:34]: 💀

File: src/routes/+page.svelte [hunk 4]
```diff
@@ -26,51 +28,30 @@
...
-</svelte:head>
-
-<h1>Welcome to Provider Dashboard</h1>
```
Comment (shannonwells) [inline src/routes/+page.svelte:37]: still need a good title though.

File: test/unit-and-integration/capacity.test.ts [hunk 5]
```diff
@@ -103,9 +103,9 @@ describe('Capacity.svelte', () => {
   });
 
   describe('if not connected', () => {
-    it('is hidden', () => {
-      const { container } = render(Capacity, { token: 'FLARP' });
-      expect(container.querySelector('div div')).toHaveClass('hidden');
+    it('is says it is not connected', () => {
```
Comment (mattheworris) [inline test/unit-and-integration/capacity.test.ts:106]:
```suggestion
    it('it says it is not connected', () => {
```

**Review by mattheworris — APPROVED** [review]
I ran the dev version with tests against a local node.
It's looking really good!
e2e tests are broken and I created a branch to see if I could fix it, but I didn't find a satisfying solution yet.

**Comment by mattheworris** [comment]
One other thing: on Rococo, my test account has 5,009.7716 XRQCY, but Provider Dashboard shows it as 50.0977. I didn't get a chance to dig in and find where that happens in the code.

File: src/lib/utils.ts [hunk 6]
```diff
@@ -54,3 +56,12 @@ export function createMailto(to: string, subject?: string, body?: string): strin
...
   return mailtoUrl.join('');
 }
+
+// Convert the provider name in hex to a human-readable value.
+export const providerNameToHuman = (name: any): string => {
+  return hexToString(name.toString());
+};
+
+export const balanceToHuman = (balance: bigint, token: string): string => {
+  return formatBalance(balance, { withSiFull: true, withUnit: token, withZero: true, decimals: 10 });
```
Comment (wilwade) [inline src/lib/utils.ts:66]:
```suggestion
  return formatBalance(balance, { withSiFull: true, withUnit: token, withZero: true, decimals: 8 });
```

**Review by wilwade — COMMENTED** [review]
One note on the decimals, but that's all from me

File: src/lib/utils.ts (hunk 6 above)
Comment (shannonwells) [reply src/lib/utils.ts]: Fixed

**Pull request merged**